### ISSUE TITLE
void-installer: DHCP request failed for WPA2-encrypted Wi-Fi network

### ISSUE THREAD

**Issue by joshuakraemer** [void-linux/void-mklive#5]
`uname -a: Linux void-live 4.12.13_1 #1 SMP PREEMPT Thu Sep 14 13:15:00 URC 2017 x86_64 GNU/Linux`

Affected program: void-installer 0.22 b4804a8

On the latest live image, the void-installer utility fails to connect to my local WPA2-encrypted Wi-Fi network. It works after restarting dhcpcd.

# Steps to reproduce this issue

1. run `sudo void-installer`
2. select "Network"
3. select the Wi-Fi interface (wlp1s0)
4. enter the details (SSID = ..., encryption = "wpa", Password = ...)
5. the following error message is shown: "ERROR: DHCP request failed for wlp1s0. Check /dev/tty8 for errors."
6. check tty8, only message is: "sending commands to master dhcpcd process"
7. select "Exit"
8. run `sudo sv restart dhcpcd`
9. repeat steps 1–4
10. the following message is shown: "Network is working properly!"

**Comment by Pitasi** [comment]
Same syntoms. The problem for me is that my wifi password contains spaces and only the first word is read from the text box when generating the wpa_supplicant config.

**Comment by LamaRaz** [comment]
i have the same problem and the suggested solution does not work
after local install however all you have to do is start wpa_supplicant and everything works...
can't get it to work during installation however

**Comment by CameronNemo** [comment]
It seems like the installer was never adapted to the dhcpcd wpa_supplicant hook being disabled (https://voidlinux.org/news/2019/01/dhcpcd-7.1.0.html).

I would recommend that:

* the vmklive dracut file be modified to enable wpa_supplicant when dhcpcd is in use
* the installer be modified to:
   * configure the network using the general config file (`/etc/wpa_supplicant/wpa_supplicant.conf`)
   * reload the running wpa_supplicant
   * remove the wpa_supplicant bits from the dhcpcd invocation

**Comment by HassanZahin** [comment]
So, I looked into the issue. Here is some problems that i've found.
1) The wpa_supplicant service isn't enabled. Without wpa_supplicant enabled, dhcpcd hook doesn't work. See https://voidlinux.org/news/2019/01/dhcpcd-7.1.0.html
2) In current iso, dhcpcd doesn't respect -w option. So even if wpa_supplicant is started manually, the installer will still throw error as it tests if it got an ip before dhcpcd leases ip.

**Comment by HassanZahin** [comment]
The possible solution is adding "wpa_supplicant -i${dev} -c${_wpasupconf%.conf}-${dev}.conf -B" in line 859 and replacing dhcpcd if doesn't respect -w in current version

**Comment by terremoth** [comment]
+1
I also have this issue.
It seems nobory cares to fix this.

**Comment by usdogu** [comment]
my solution : 
execute
```
wpa_passphrase <MYSSID> <passphrase> >> /etc/wpa_supplicant/wpa_supplicant-<device_name>.conf

```
(you can learn your device name by executing ```ip a```)
then run
``` 
ln -s /etc/sv/wpa_supplicant /var/service/
```

then start the void installer

**Comment by timsofteng** [comment]
Bug still exists.

**Comment by ndgnuh** [comment]
I had to unblock the wifi with `rfkill` first. After that, everything worked.
```
rfkill unblock wlan
```

**Comment by terremoth** [comment]
> I had to unblock the wifi with `rfkill` first. After that, everything worked.
> 
> ```
> rfkill unblock wlan
> ```

I had to do that once too...
Seems nobody cares about wifi/wireless things on these minimal distros. Also seems they test only internet directly on ethernet cable before launch new distros versions. I am not speaking only about Void here. I had these problems even in distros such as Alpine, Adelie and AntiX. 

The truth is: already passed the time to replace `wpa_supplicant` and NetworkManager sh1ts to newer and less RAM consuming softwares (nm-applet is a RAM cancer). Very old and bloated softwares, sometimes with unexpected behavior and not easy/fast to configure.

**Comment by nonetrix** [comment]
Same here

**Comment by henrikstroem** [comment]
> my solution : execute
> 
> ```
> wpa_passphrase <MYSSID> <passphrase> >> /etc/wpa_supplicant/wpa_supplicant-<device_name>.conf
> ```
> 
> (you can learn your device name by executing `ip a`) then run
> 
> ```
> ln -s /etc/sv/wpa_supplicant /var/service/
> ```
> 
> then start the void installer

A few more steps was needed for this to work in my case. My network adapter is called wlp3s0.

> ln -s /etc/sv/wpa_supplicant /etc/runit/runsvdir/default

This will start the service. To see any error messages:

> wpa_cli -i wlp3s0

If you don't see any errors running this command, your network is probably ok, and you can run the void installer agian.

**Comment by terremoth** [comment]
> > my solution : execute
> > ```
> > wpa_passphrase <MYSSID> <passphrase> >> /etc/wpa_supplicant/wpa_supplicant-<device_name>.conf
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > (you can learn your device name by executing `ip a`) then run
> > ```
> > ln -s /etc/sv/wpa_supplicant /var/service/
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > then start the void installer
> 
> A few more steps was needed for this to work in my case. My network adapter is called wlp3s0.
> 
> > ln -s /etc/sv/wpa_supplicant /etc/runit/runsvdir/default
> 
> This will start the service. To see any error messages:
> 
> > wpa_cli -i wlp3s0
> 
> If you don't see any errors running this command, your network is probably ok, and you can run the void installer agian.

Whattttttttttt, they haven't resolved this yet? half of 2022?

**Comment by terremoth** [comment]
Please someone tag void devs here, this is a easy issue to solve, just adding commands to the installer...

**Comment by Vaelatern** [comment]
PR welcome

**Comment by CameronNemo** [comment]
@Vaelatern I made a PR and linked it from this ticket. Testers welcome.

**Comment by 3BD-R** [comment]
Work around using chroot method

- `ip -c a`
- `wpa_supplicant -B -i wlp3s0 -c /etc/wpa_supplicant/wpa_supplicant.conf`
  > Successfully initialized wpa_supplicant
- `wpa_cli`
  - `add_network`
  >0
  - `set_network 0 ssid "hidden_ssid"`
  - `set_network 0 psk "password"`
  - `set_network 0 scan_ssid 1`
    > CONNECT TO HIDDEN SSID
  - `enable_network 0`
  - `save_config`
  - `select_network 0`

**Comment by 3BD-R** [comment]
This solution did not work for me at the begainning so i did that:

- `ip -c a`
- `wpa_supplicant -B -i wlp3s0 -c /etc/wpa_supplicant/wpa_supplicant.conf`
  > Successfully initialized wpa_supplicant
- `wpa_cli`
  - `add_network`
  >0
  - `set_network 0 ssid "hidden_ssid"`
  - `set_network 0 psk "password"`
  - `set_network 0 scan_ssid 1`
    > CONNECT TO HIDDEN SSID
  - `enable_network 0`
  - `save_config`
  - `select_network 0`
  Then i just realised that issue happened because of "hidden wifi" so this solution works but you have to add `scan_ssid=1` inside network block befor `ln -s /etc/.....`
> my solution : execute
> 
> ```
> wpa_passphrase <MYSSID> <passphrase> >> /etc/wpa_supplicant/wpa_supplicant-<device_name>.conf
> ```
> 
> (you can learn your device name by executing `ip a`) then run
> 
> ```
> ln -s /etc/sv/wpa_supplicant /var/service/
> ```
> 
> then start the void installer

**Comment by skyler544** [comment]
Do I need to make a fresh live iso in order to benefit from this fix? This issue is not resolved when using the latest iso from the Void homepage.

**Comment by classabbyamp** [comment]
yes, until the new iso is released

**Comment by 0x8ff** [comment]
Happy to find that this issue has been fixed, nearly a year later :)

**Comment by paper42** [comment]
4 years actually 🙂

**Comment by NovaFugaz** [comment]
I really can't believe there's been 6 years and this problem still persists.

**Comment by classabbyamp** [comment]
then make a new issue with more info
